### ISSUE TITLE
look at how I3XXXRandomServiceFactory cascades (Trac #2124)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2116]
Freshly installed Ubuntu 18 bot w/o `sprng2` or `root` had several tests looking for I3SPRNGRandomServiceFactory. Those tests or projects shouldn't build if `sprng2` is required, but not detected. Is `sprng2` the last choice for rng's? Should we maybe have a SF based on the system rng, if only for testing purposes?

Migrated from https://code.icecube.wisc.edu/ticket/2124
```json
{
    "status": "closed", 
    "changetime": "2019-03-18T21:47:49", 
    "description": "Freshly installed Ubuntu 18 bot w/o `sprng2` or `root` had several tests looking for I3SPRNGRandomServiceFactory. Those tests or projects shouldn't build if `sprng2` is required, but not detected. Is `sprng2` the last choice for rng's? Should we maybe have a SF based on the system rng, if only for testing purposes?", 
    "reporter": "nega", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1552945669007619", 
    "component": "combo simulation", 
    "summary": "look at how I3XXXRandomServiceFactory cascades", 
    "priority": "normal", 
    "keywords": "sprng2", 
    "time": "2017-12-07T19:19:01", 
    "milestone": "Vernal Equinox 2019", 
    "owner": "nega", 
    "type": "cleanup"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20171208T10:07:35: kjmeagher commented:

> switching to `I3GSLRandomService` should be appropriate for all unit tests

**Comment by icecube-trac** [comment]
Trac update at 20180922T09:01:48: olivas changed milestone from "" to "Winter Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20190318T21:47:49: 
* kjmeagher changed status from "new" to "closed"
* kjmeagher changed resolution from "" to "fixed"
* kjmeagher commented:

> In changeset 172126:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="172126"
> move all instances of I3SPRNGRandomService to I3GSLRandomService since SPRNG is optional,
> disabled simprod-scripts tests if SPRNG is not found since SPRNG is required for simprod scripts
> closes #2116
> ```
